test/fixtures/comprehensions/assignment-expr/actual.js

```diff
@@ -0,0 +1,3 @@
+[for idx i in Array(10):
+  now x = f(i)
```

Comment:
Cool. Yeah it's there in `addImplicitReturns` from before that was moved to an `exit()` visitor, when `x = 3` was going to turn into `const x = 3`. It actually shouldn't be needed for returns anymore, but now we're relying on it for this. Fine to leave as-is I think.